plane/src/database/backend.rs

```diff
@@ -181,29 +181,53 @@ impl<'a> BackendDatabase<'a> {
         &self,
         backend: &BackendName,
         state: BackendState,
-    ) -> sqlx::Result<()> {
+    ) -> sqlx::Result<bool> {
         let mut txn = self.db.pool.begin().await?;
 
         emit_with_key(&mut *txn, &backend.to_string(), &state).await?;
 
-        sqlx::query!(
+        let last_status = state.status();
+        let last_status_number = last_status.as_int();
```

Comment:
nit: can we call these vars `new_status` and `new_status_number` or `next_status`?